src/astUtils/reflection.ts

```diff
@@ -191,56 +195,48 @@ export function isFunctionParameterExpression(element: Expression | Statement):
     return element?.constructor.name === 'FunctionParameterExpression';
 }
 
-// Value/Type reflection
-
-export function isInvalid(value: any): value is BrsInvalid {
-    return value?.kind === ValueKind.Invalid;
-}
-export function isBoolean(value: any): value is BrsBoolean {
-    return value?.kind === ValueKind.Boolean;
+// BscType reflection
+export function isStringType(value: any): value is StringType {
+    return value?.constructor.name === StringType.name;
 }
-export function isString(value: any): value is BrsString {
-    return value?.kind === ValueKind.String;
+export function isFunctionType(e: any): e is FunctionType {
+    return e?.constructor.name === FunctionType.name;
 }
-export function isNumber(value: any): value is BrsNumber {
-    return value && (
-        value.kind === ValueKind.Int32 ||
-        value.kind === ValueKind.Int64 ||
-        value.kind === ValueKind.Float ||
-        value.kind === ValueKind.Double
-    );
+export function isBooleanType(e: any): e is BooleanType {
+    return e?.constructor.name === BooleanType.name;
 }
-export function isStringType(value: any): value is StringType {
-    return value?.constructor.name === 'StringType';
+export function isIntegerType(e: any): e is IntegerType {
+    return e?.constructor.name === IntegerType.name;
 }
-
-export function isFunctionType(e: any): e is FunctionType {
-    return e?.constructor.name === 'FunctionType';
+export function isLongIntegerType(e: any): e is LongIntegerType {
+    return e?.constructor.name === LongIntegerType.name;
 }
-export function isRoArray(e: any): e is RoArray {
-    return e?.constructor.name === 'RoArray';
+export function isFloatType(e: any): e is FloatType {
+    return e?.constructor.name === FloatType.name;
 }
-export function isRoAssociativeArray(e: any): e is RoAssociativeArray {
-    return e?.constructor.name === 'RoAssociativeArray';
+export function isDoubleType(e: any): e is DoubleType {
+    return e?.constructor.name === DoubleType.name;
 }
-export function isRoSGNode(e: any): e is RoSGNode {
-    return e?.constructor.name === 'RoSGNode';
+export function isNumberType(e: any): e is IntegerType | LongIntegerType | FloatType | DoubleType {
+    return [
+        IntegerType.name,
+        LongIntegerType.name,
+        FloatType.name,
+        DoubleType.name
+    ].includes(e?.constructor.name);
```

Comment:
I don't think that's better than using a `kind` property. The new types don't have them?
Also this creates a temporary array so it's not a great runtime profile - this could hoisted as a const outside the function.

Reply:
BrighterScript will eventually support more advanced types. `kind` only works for the built-in BrightScript types. For example, I would like to add typescript-like interfaces. So we would have something like:
```
new InterfaceType([{
    name: 'name',
    type: new StringType()
}, {
    name: 'age',
    type: new IntegerType()
}])
```

There's no way to represent that with a `kind` property. The closest we get is `Object` which will not provide enough the level of type safety I'd like to eventually enforce in BrighterScript. 

But good point about the runtime array. I'll hoist that out into a const.